### PR TITLE
Added scraper for IrishCentral

### LINKED ISSUE #1167: IrishCentral
if you could add support for this, it would be fantastic.
https://www.irishcentral.com/culture/food-drink/shepherds-pie-recipe

### PR DESCRIPTION
**Pull request by bencondemi**
Resolves #1167 

With ingredients in "ul" structure and no yields ~ https://www.irishcentral.com/culture/food-drink/shepherds-pie-recipe
With ingredients in "p" structure and with yields ~ https://www.irishcentral.com/culture/food-drink/apple-jameson-tart-recipe

### CODE REVIEW

**Comment by bencondemi** [comment]
@jknndy, at your convenience, please let me know your thoughts!

File: recipe_scrapers/irishcentral.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        description = self.soup.find('meta', {'property': 'og:description'})['content']
+        return description if description else FieldNotProvidedByWebsiteException(return_value=None)
+
+    def image(self):
+        image = self.soup.find('meta', {'property': 'og:image'})['content']
+        return image if image else FieldNotProvidedByWebsiteException(return_value=None)
+
+    def ingredients(self):
+        # Check if the ingredients are in a <p> structure (https://www.irishcentral.com/culture/food-drink/apple-jameson-tart-recipe)
+        ingredients_label = self.soup.find("p", text=re.compile(r"Ingredients:"))
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:36]:
```suggestion
        ingredients_label = self.soup.find("p", string=re.compile(r"Ingredients:"))
```
`DeprecationWarning: The 'text' argument to find()-type methods is deprecated. Use 'string' instead.`

File: recipe_scrapers/irishcentral.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    for li in ingredients_list.find_all("li")
+                    if li.get_text(strip=True) and li.get_text(strip=True).lower() not in ["most read", "popular"]
+                ]
+                if ingredients_list:
+                    return ingredients_list
+
+        return FieldNotProvidedByWebsiteException(return_value=None)
+
+    def instructions(self):
+        instructions_label = self.soup.find("p", text=re.compile(r"Method:"))
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:70]:
```suggestion
        instructions_label = self.soup.find("p", string=re.compile(r"Method"))
        instructions_list = []
```
`DeprecationWarning: The 'text' argument to find()-type methods is deprecated. Use 'string' instead.` & moving initialization of instructions_list to the top of the method

File: recipe_scrapers/irishcentral.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                    break
+
+                instructions_list.append(instruction_text)
+
+            if not instructions_list:
+                return FieldNotProvidedByWebsiteException(return_value=None)
+
+            return "\n".join(instructions_list)
+
+        return FieldNotProvidedByWebsiteException(return_value=None)
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:89]:
```suggestion
                if instruction_text and not instruction_text.startswith("*"):
                    instructions_list.append(instruction_text)
                else:
                    break

        return "\n".join(instructions_list)
```

File: recipe_scrapers/irishcentral.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    def title(self):
+        title = self.soup.find('meta', {'property': 'og:title'})['content']
+        return title if title else FieldNotProvidedByWebsiteException(return_value=None)
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:93]:
```suggestion
        title = self.soup.find("meta", {"property": "og:title"})["content"]
        return title
```

File: recipe_scrapers/irishcentral.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+    def total_time(self):
+        return FieldNotProvidedByWebsiteException(return_value=None)
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:96]:
```suggestion
        raise FieldNotProvidedByWebsiteException(return_value=None)
```

File: recipe_scrapers/irishcentral.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+    def yields(self):
+        serves_label = self.soup.find("strong", text=lambda t: t and "Serves:" in t)
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:99]:
```suggestion
        serves_label = self.soup.find("p", string=re.compile(r"Serves\s+\d+"))
```
I think using `re` would be a better fit allowing for some slight variation on the accepted text.

File: recipe_scrapers/irishcentral.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+        if serves_label:
+            serves_text = normalize_string(serves_label.get_text())
+            serves_value = serves_text.replace("Serves:", "").strip()
+            return f"Serves {serves_value}"
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:104]:
```suggestion
            serves_value = re.search(r"Serves\s+(\d+)", serves_text).group(1)
            return get_yields(serves_value)
```
Another `re` usage & passes the output at our built in util get_yields which handles the processing and appending Servings to the output.

File: recipe_scrapers/irishcentral.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+        raise FieldNotProvidedByWebsiteException(return_value=None)
+
+    def keywords(self):
+        keywords = self.soup.find('meta', {'name': 'keywords'})['content']
+        if keywords:
+            return [keyword.strip() for keyword in keywords.split(',')]
+        else:
+            raise FieldNotProvidedByWebsiteException(return_value=None)
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:113]:
```suggestion
        keywords = self.soup.find("meta", {"name": "keywords"})["content"]
        return [keyword.strip() for keyword in keywords.split(",")] if keywords else []
```

File: tests/test_data/irishcentral.com/irishcentral.json [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    "Pour the remaining cream into a bowl. Place over simmering water. Stirring constantly, and continue to cook until the custard thickens. Set aside - keep warm.",
+    "Serving suggestions",
+    "Serve the tart, dusted with icing sugar, with the warm custard. Vanilla ice cream, thin almond biscuits, raspberries, etc. are optional.",
+    "John Howard was quite possibly Ireland's first celebrity chef. For 25 years he and his wife, Catherine, ran Le Coq Hardi restaurant, in Dublin 4, in 1977. It was one of the best-known dining locations in the country. Le Coq Hardi was known for its great food, expensive wine and for being where the captains of industry dined on lavish expense accounts mixed in with a little hanky panky.",
+    "In early 2000, he retired, sold the premises to a wine company, and moved to Spain to consult with Bord Bia."
+  ],
+  "yields": "Serves 4",
+  "description": "Irish celebrity chef John Howard's apple and Jameson tart recipe marries two Irish favorites.",
+  "total_time": "FieldNotProvidedByWebsiteException('Suggested return value None is not from recipe source.')",
```
Comment (jknndy) [inline tests/test_data/irishcentral.com/irishcentral.json:29]:
```suggestion
  "total_time": null,
```

File: recipe_scrapers/irishcentral.py [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+            if ingredients_list:
+                ingredients_list = [
+                    normalize_string(li.get_text())
+                    for li in ingredients_list.find_all("li")
+                    if li.get_text(strip=True) and li.get_text(strip=True).lower() not in ["most read", "popular"]
+                ]
+                if ingredients_list:
+                    return ingredients_list
+
+        return FieldNotProvidedByWebsiteException(return_value=None)
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:67]:
```suggestion
```

File: recipe_scrapers/irishcentral.py [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+    def yields(self):
+        serves_label = self.soup.find("strong", text=lambda t: t and "Serves:" in t)
+
+        if serves_label:
+            serves_text = normalize_string(serves_label.get_text())
+            serves_value = serves_text.replace("Serves:", "").strip()
+            return f"Serves {serves_value}"
+
+        raise FieldNotProvidedByWebsiteException(return_value=None)
```
Comment (jknndy) [inline recipe_scrapers/irishcentral.py:106]:
```suggestion
```

**Review by jknndy — CHANGES_REQUESTED** [review]
Thanks for the contribution! Made some comments throughout, some of these changes will require test updates. Also once you're done with these be sure to run our test suite locally and the formatted as well. Let me know if you have any questions!
```
black .\recipe_scrapers\irishcentral.py && python -m unittest
``` 

The exception `FieldNotProvidedByWebsiteException` should only be used when the site never provides the field and should only be used for mandatory fields (not an issue here just mentioning). So there should only be one implementation for total_time in this instance.

File: recipe_scrapers/irishcentral.py (hunk 6 above)
Comment (bencondemi) [reply recipe_scrapers/irishcentral.py:84]: > serves_label = self.soup.find("strong", text=lambda t: t and "Serves:" in t)

@jknndy, I tried this suggestion (and the next one as well); however, it was unable to find the correct fields in https://www.irishcentral.com/culture/food-drink/apple-jameson-tart-recipe, so I reverted back to my original.

With that being said, I ran the formatter on the .py file and corrected some of the inconsistent test cases. Please let me know your thoughts.

**Comment by bencondemi** [comment]
> Thanks for the contribution! Made some comments throughout, some of these changes will require test updates. Also once you're done with these be sure to run our test suite locally and the formatter as well. Let me know if you have any questions!
> 
> ```
> black .\recipe_scrapers\irishcentral.py && python -m unittest
> ```
> 
> The exception `FieldNotProvidedByWebsiteException` should only be used when the site never provides the field and should only be used for mandatory fields (not an issue here just mentioning). So there should only be one implementation for total_time in this instance.

@jknndy, I ran the formatter on the .py file, revised with your suggestions, and corrected some of the inconsistent test cases. Please let me know your thoughts.

**Comment by bencondemi** [comment]
@jknndy, I am having trouble passing the linter (seemingly due to isort inconsistencies within __init__.py). 

I have ran `isort recipe_scrapers/__init__.py` and am wondering if you might be able to help me understand what else I can do to fix this? Thanks!

File: recipe_scrapers/__init__.py [hunk 12]
```diff
@@ -23,14 +23,10 @@
...
-    FieldNotProvidedByWebsiteException,
-    NoSchemaFoundInWildMode,
-    RecipeSchemaNotFound,
-    StaticValueException,
-    WebsiteNotImplementedError,
-)
+from ._exceptions import (ElementNotFoundInHtml,
+                          FieldNotProvidedByWebsiteException,
+                          NoSchemaFoundInWildMode, RecipeSchemaNotFound,
+                          StaticValueException, WebsiteNotImplementedError)
```
Comment (jknndy) [inline recipe_scrapers/__init__.py:29]:
```suggestion
from ._exceptions import (
    ElementNotFoundInHtml,
    FieldNotProvidedByWebsiteException,
    NoSchemaFoundInWildMode,
    RecipeSchemaNotFound,
    StaticValueException,
    WebsiteNotImplementedError,
)
```

**Review by jknndy — APPROVED** [review]

**Comment by jknndy** [comment]
Thanks again for the contribution @bencondemi !

**Pull request merged**